### DISCUSSION TITLE
How to set up env varraible for google analytics

### DISCUSSION THREAD

**Discussion by tvdboom** [squidfunk/mkdocs-material#3638] [Q&A]
Hi,

I want to add Google Analytics tracking to `mkdocs.yml` but I don't want to publsih my token. After some research I saw that you do it yourself like this:
```property: !ENV GOOGLE_ANALYTICS_KEY```

but you also asked this same question [here](https://github.com/mkdocs/mkdocs/issues/1195) (although this was in 2017) and answered that it should be done like this:
``` - !!python/object/apply:os.getenv ["GOOGLE_ANALYTICS_TOKEN"]```

Which one should I use and where do I store the actual token? I know I can add it to GitHub Secrets but not sure if `!ENV` is going to read from there.

**Comment by jaceklaskowski** [comment]
It should now be as follows:

```
property: !ENV GOOGLE_ANALYTICS_KEY
```

**Reply by tvdboom** [reply]
And where do I store the actual value of the key?

**Reply by jorge-campo** [reply]
Hey @tvdboom, just saw your question while looking for another thing.

I guess you can use:
- GitHub secrets to store the Google Analytics key value (see [Encrypted secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets)).
- The `env:` environment variable in the mkdocs.yml configuration file (see the example for the `GH_TOKEN` variable in the Material for MkDocs docs [here](https://squidfunk.github.io/mkdocs-material/publishing-your-site/#with-github-actions)).

**Reply by tvdboom** [reply]
Hi, thanks for answering. The example for the `GH_TOKEN` variable in the Material for MkDocs docs [here](https://squidfunk.github.io/mkdocs-material/publishing-your-site/#with-github-actions) refers to a GitHub Action, not the `mkdocs.yml` file. I don't use GitHub Action to deploy the website. How can I access the secret from the `mkdocs.yml` file?

**Reply by jorge-campo** [reply]
You're right! The example refers to a GitHub Action; I'm sorry for mixing these two things.

We don't use GitHub Actions either; we use Jenkins. And we use a Jenkins environment variable for passing the `GH_TOKEN` value to the build script. If this is not your case, this discussion may be helpful: https://github.com/squidfunk/mkdocs-material/discussions/3863#discussioncomment-2668461.
